### PR TITLE
update models to contain sortable data types

### PR DESCRIPTION
**Pull request by morkusporkus**

### CODE REVIEW

**Review by benjaminsampica — CHANGES_REQUESTED** [review]
Don’t duplicate the values just pass the data type instead and format it on the client.

additionally, the tests didn’t pass.

File: src/Client/Features/Players/List.razor [hunk 1]
```diff
@@ -44,8 +44,8 @@
         </MudTd>
         <MudTd DataLabel=@nameof(PlayerItem.Position)>@context.Position</MudTd>
         <MudTd DataLabel=@nameof(PlayerItem.Team)>@context.Team</MudTd>
-        <MudTd DataLabel="Contract Value">@context.ContractValue</MudTd>
-        <MudTd DataLabel="Contract Length">@context.YearContractExpires</MudTd>
+        <MudTd DataLabel="Contract Value">@context.ContractValue.ToString("C0")</MudTd>
+        <MudTd DataLabel="Contract Length">@context.YearContractExpires.ToString()</MudTd>
```
Comment (benjaminsampica) [inline src/Client/Features/Players/List.razor:48]: Is .ToString here superfluous?

File: src/Client/Features/FreeAgents/List.razor [hunk 2]
```diff
@@ -25,8 +25,8 @@
                 <MudTh><MudTableSortLabel SortBy="new Func<FreeAgentItem,object>(x=> x.Name)">Name</MudTableSortLabel></MudTh>
                 <MudTh><MudTableSortLabel SortBy="new Func<FreeAgentItem,object>(x=> x.Position)">Position</MudTableSortLabel></MudTh>
                 <MudTh><MudTableSortLabel SortBy="new Func<FreeAgentItem,object>(x=> x.Team)">Former Team</MudTableSortLabel></MudTh>
-                <MudTh><MudTableSortLabel SortBy="new Func<FreeAgentItem,object>(x=> x.BiddingEnds)">Bidding Ends</MudTableSortLabel></MudTh>
-                <MudTh><MudTableSortLabel SortBy="new Func<FreeAgentItem,object>(x=> x.HighestBid)">Highest Bid</MudTableSortLabel></MudTh>
+                <MudTh><MudTableSortLabel SortBy="new Func<FreeAgentItem,object>(x=> x.BiddingEndDate)">Bidding Ends</MudTableSortLabel></MudTh>
+                <MudTh><MudTableSortLabel SortBy="new Func<FreeAgentItem,object>(x=> x.HighestBidValue)">Highest Bid</MudTableSortLabel></MudTh>
```
Comment (benjaminsampica) [inline src/Client/Features/FreeAgents/List.razor:29]: “Value” doesn’t add anything to me IMO, thoughts?

File: src/Server/Features/FreeAgents/List.cs [hunk 3]
```diff
@@ -77,10 +78,14 @@ public ListMappingProfile()
                 s.Bids.GetHighestBidder().TeamId == currentUserTeamId)
             )
             .ForMember(d => d.BiddingEnds, mo => mo.MapFrom(s => s.EndOfFreeAgency!.Value.ToShortDateString()))
+            .ForMember(d => d.BiddingEndDate, mo => mo.MapFrom(s => s.EndOfFreeAgency))
```
Comment (benjaminsampica) [inline src/Server/Features/FreeAgents/List.cs:81]: I don’t think having two properties with different formatting is the way to go. Thoughts?

File: src/Client/Features/FreeAgents/List.razor (hunk 2 above)
Comment (morkusporkus) [reply src/Client/Features/FreeAgents/List.razor:29]: oops thgought i deleted this

File: src/Server/Features/FreeAgents/List.cs (hunk 3 above)
Comment (morkusporkus) [reply src/Server/Features/FreeAgents/List.cs:81]: Agree, I tried to remove them, I guess I am just flaking

File: src/Client/Features/Players/List.razor (hunk 1 above)
Comment (morkusporkus) [reply src/Client/Features/Players/List.razor:48]: yea

**Review by benjaminsampica — APPROVED** [review]

**Pull request merged**